unlock-app/src/components/interface/checkout/main/index.tsx

```diff
@@ -39,7 +39,7 @@ export function Checkout({
     },
   })
   const [state] = useActor(checkoutService)
-  const { account } = useAuth()
+  const { account, network: connectedNetwork } = useAuth()
```

Comment:
A more robust way would be to listen to provider.on change network events so any unnecessary state doesn't cause re-render and back to select. WDYT?

Reply:
yeah! that's a great idea! changed